xbmc/interfaces/json-rpc/schema/methods.json

```diff
@@ -698,14 +707,19 @@
       { "name": "sort", "$ref": "List.Sort" },
       { "name": "filter",
         "type": [
-          { "type": "object", "properties": { "genreid": { "$ref": "Library.Id", "required": true } }, "additionalProperties": false },
-          { "type": "object", "properties": { "genre": { "type": "string", "minLength": 1, "required": true } }, "additionalProperties": false },
+          { "type": "object", "properties": { "genreid": { "$ref": "Library.Id", "required": true, "description": "Song genre. Filter for existance of songs with this genre" } }, "additionalProperties": false },
+          { "type": "object", "properties": { "genre": { "type": "string", "minLength": 1, "required": true, "description": "Song genre. Filter for existance of songs with this genre" } }, "additionalProperties": false },
           { "type": "object", "properties": { "artistid": { "$ref": "Library.Id", "required": true } }, "additionalProperties": false },
+          { "type": "object", "properties": { "artistid": { "$ref": "Library.Id", "required": true }, "roleid": { "$ref": "Library.Id", "required": true }}, "additionalProperties": false },
+          { "type": "object", "properties": { "artistid": { "$ref": "Library.Id", "required": true }, "role": { "type": "string", "minLength": 1, "required": true }}, "additionalProperties": false },
           { "type": "object", "properties": { "artist": { "type": "string", "minLength": 1, "required": true } }, "additionalProperties": false },
+          { "type": "object", "properties": { "artist": { "type": "string", "minLength": 1, "required": true }, "roleid": { "$ref": "Library.Id", "required": true }}, "additionalProperties": false },
+          { "type": "object", "properties": { "artist": { "type": "string", "minLength": 1, "required": true }, "role": { "type": "string", "minLength": 1, "required": true }}, "additionalProperties": false },
           { "$ref": "List.Filter.Albums" }
         ]
       },
-      { "name": "includesingles", "type": "boolean", "default": false }
+      { "name": "includesingles", "type": "boolean", "default": false },
+      { "name": "allroles", "$ref": "Optional.Boolean", "description": "Whether or not to include all roles when filtering by artist, rather than default of excluding other contributors. If the parameter is not passed or is passed as null then it is taken as false. When true it overrides any role filter value." }
```

Comment:
This parameter only makes sense in combination with the `artist` or `artistid` filters right? Can't this be achieved through those filter's default value?